src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -157,6 +161,157 @@ namespace OpenMS
       }
     }
   }
+  
+  void MRMTransitionGroupPicker::calculatePeakApexInt_(const MSChromatogram& chromatogram,
+    double best_left, double best_right, 
+    ConvexHull2D::PointArrayType & hull_points,
+    double & intensity_sum, 
+    double & rt_sum,
+    double & peak_apex_int,
+    double & peak_apex_rt)
+  {  
+    intensity_sum = 0.0;
+    rt_sum = 0.0;
+    peak_apex_int = -1;
+    double peak_apex_dist = std::fabs(chromatogram.begin()->getMZ() - peak_apex_rt);
+    // FEATURE : use RTBegin / MZBegin -> for this we need to know whether the template param is a real chromatogram or a spectrum!
+    for (MSChromatogram::const_iterator it = chromatogram.begin(); it != chromatogram.end(); ++it)
+    {
+      if (it->getMZ() > best_left && it->getMZ() < best_right)
+      {
+        DPosition<2> p;
+        p[0] = it->getMZ();
+        p[1] = it->getIntensity();
+        hull_points.push_back(p);
+        if (std::fabs(it->getMZ() - peak_apex_rt) <= peak_apex_dist)
+        {
+          peak_apex_int = p[1];
+          peak_apex_dist = std::fabs(it->getMZ() - peak_apex_rt);
+        }
+        rt_sum += it->getMZ();
+        intensity_sum += it->getIntensity();
+      }
+    }
+  }
+  
+  void MRMTransitionGroupPicker::calculatePeakShapeMetrics_(const MSChromatogram& chromatogram, 
+    double best_left, double best_right, 
+    double peak_height, double peak_apex_rt, double avg_noise_level,
+    PeakShapeMetrics_ & peakShapeMetrics)
+  {
+    peakShapeMetrics.points_across_baseline = 0;
+    double start_intensity, end_intensity;
+    double intensity, intensity_prev, retention_time, retention_time_prev;
+    double delta_rt, delta_int, height_5, height_10, height_50;
+    
+    // pass 1
+    for (MSChromatogram::const_iterator it = chromatogram.begin() + 1; it != chromatogram.end(); ++it)
+    {
+      MSChromatogram::const_iterator it_prev = it;
+      --it_prev; //previous point
+      intensity = std::max(it->getIntensity()-avg_noise_level, 0.0); //background-subtracted intensity
+      intensity_prev = std::max(it_prev->getIntensity()-avg_noise_level, 0.0); //background-subtracted intensity of the previous point
+      retention_time = it->getMZ();
+      retention_time_prev = it_prev->getMZ();
+
+      // start and end intensities
+      if (retention_time_prev == best_left)
```

Comment:
I am a bit uncomfortable with comparing floats, this has a tendency to go wrong ...